examples/Cassie/run_pd_controller.cc

```diff
@@ -1,43 +1,78 @@
+#include <gflags/gflags.h>
+
+#include "systems/primitives/output_subvector_pass_through.h"
+#include "drake/systems/analysis/simulator.h"
 #include "drake/systems/framework/diagram_builder.h"
+#include "drake/systems/lcm/lcm_interface_system.h"
 #include "drake/systems/lcm/lcm_publisher_system.h"
 #include "drake/systems/lcm/lcm_subscriber_system.h"
-#include "drake/systems/lcm/lcm_interface_system.h"
-#include "drake/systems/analysis/simulator.h"
 
-#include "dairlib/lcmt_robot_output.hpp"
-#include "dairlib/lcmt_robot_input.hpp"
 #include "dairlib/lcmt_pd_config.hpp"
-#include "systems/robot_lcm_systems.h"
+#include "dairlib/lcmt_robot_input.hpp"
+#include "dairlib/lcmt_robot_output.hpp"
+#include "examples/Cassie/cassie_utils.h"
 #include "systems/controllers/linear_controller.h"
 #include "systems/controllers/pd_config_lcm.h"
-#include "examples/Cassie/cassie_utils.h"
+#include "systems/robot_lcm_systems.h"
 
 namespace dairlib {
 
-using drake::systems::lcm::LcmSubscriberSystem;
-using drake::systems::lcm::LcmPublisherSystem;
-using drake::multibody::MultibodyPlant;
 using drake::geometry::SceneGraph;
+using drake::multibody::MultibodyPlant;
 using drake::systems::DiagramBuilder;
+using drake::systems::lcm::LcmPublisherSystem;
+using drake::systems::lcm::LcmSubscriberSystem;
 
+// lcm channels
+DEFINE_string(channel_x, "CASSIE_STATE_DISPATCHER",
+              "LCM channel for receiving state. "
+              "Use CASSIE_STATE_SIMULATION to get state from simulator, and "
+              "use CASSIE_STATE_DISPATCHER to get state from state estimator");
+DEFINE_string(channel_u, "CASSIE_INPUT",
+              "The name of the channel which publishes command");
+
+// Cassie model parameter
+DEFINE_bool(floating_base, true, "Fixed or floating base model");
+
+int doMain(int argc, char* argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
 
-int doMain() {
   DiagramBuilder<double> builder;
   DiagramBuilder<double> builder_null;
 
   auto lcm = builder.AddSystem<drake::systems::lcm::LcmInterfaceSystem>(
       "udpm://239.255.76.67:7667?ttl=0");
 
   SceneGraph<double>& scene_graph = *builder.AddSystem<SceneGraph>();
+  SceneGraph<double>& scene_graph_null = *builder_null.AddSystem<SceneGraph>();
   scene_graph.set_name("scene_graph");
 
-  MultibodyPlant<double>& plant =
+  MultibodyPlant<double>& plant_fixed_base =
```

Comment:
I thought I had added this comment to the last review, though maybe I deleted by mistake. Why does the controller need two plants? I can see a need to determine whether the controlled plant is floating or fixed, but requiring both seems unnecessary.

Reply:
Your comment is still in this PR. Here is my reply to that comment: 
https://github.com/DAIRLab/dairlib/pull/130#discussion_r407109605

Reply:
After discussing with @mposa , I realized that `PDConfigReceiver` already takes care of the floating base part of the model when creating the gain matrix K. Therefore there is no need for having two plants.

Reply:
I agree that it's not necessary. I added it as an easy way to get the number of positions/velocities for each plant to determine how "filter" out the floating base states for the pd controller.